helmchart/sdp/values.yaml

```diff
@@ -4,116 +4,178 @@
-    # domain: "localhost:8000"
     domain: sdp.localhost.com
     port: "8000"
     metricsPort: "8002"
-  # =========================== START sdp.image ===========================
+
+  ## @extra sdp.image Configuration related to the Docker image used by the SDP service.
+  ## @param sdp.image.repository Docker image repository for the SDP backend service.
+  ## @param sdp.image.pullPolicy Image pull policy for the SDP service. For locally built images, consider using "Never" or "IfNotPresent".
+  ## @param sdp.image.tag Docker image tag for the SDP service. If set, this overrides the default value from `.Chart.AppVersion`.
   image:
-    # replace the `repository` with the actual image name.
     repository: stellar/stellar-disbursement-platform-backend
-    # for locally built images, use `pullPolicy: Never or IfNotPresent`.
     pullPolicy: IfNotPresent
-    # If `tag` is set, it'll override the default value set in `.Chart.AppVersion`.
     tag: "latest"
 
-  # =========================== START sdp.deployment ===========================
+  ## @extra sdp.deployment Configuration related to the deployment of the SDP service.
+  ## @param sdp.deployment.annotations Annotations to be added to the deployment.
+  ## @extra sdp.deployment.podAnnotations Annotations specific to the pods, mainly for integration with monitoring tools like Prometheus.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/path The path where Prometheus metrics are exposed.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/port The port on which Prometheus metrics are exposed. This should match the SDP metrics port.
+  ## @param sdp.deployment.podAnnotations.prometheus.io/scrape Indicates if Prometheus should scrape metrics from this pod.
+  ## @param sdp.deployment.podSecurityContext Security settings for the pods.
+  ## @param sdp.deployment.securityContext Security settings for the container within the pod.
+  ## @extra sdp.deployment.strategy Configuration related to the deployment strategy, ensuring smooth updates and minimal downtime.
+  ## @param sdp.deployment.strategy.type Specifies the deployment strategy, RollingUpdate is preferred for minimal downtime.
+  ## @param sdp.deployment.strategy.rollingUpdate.maxUnavailable The maximum number of pods that can be unavailable during an update.
+  ## @param sdp.deployment.strategy.rollingUpdate.maxSurge The maximum number of pods that can be created over the desired number of pods.
+  ## @param sdp.deployment.nodeSelector Node selector to determine which nodes should run the pods.
+  ## @param sdp.deployment.tolerations Tolerations to ensure pods aren't scheduled on unsuitable nodes.
+  ## @param sdp.deployment.affinity Affinity rules to determine where pods get scheduled based on node conditions.
   deployment:
     annotations:
-
     podAnnotations:
       prometheus.io/path: /metrics
       prometheus.io/port: '{{ include "sdp.metricsPort" . }}'
       prometheus.io/scrape: "true"
```

Comment:
💡  I believe podAnnotations should also default to nil or { }. These prometheus-related ones are implementation specific.

The same applies for all:
- [ ] sdp.deployment.podAnnotations
- [ ] anchorPlatform.deployment.podAnnotations
- [ ] tss.deployment.podAnnotations
- [ ] dashboard.deployment.podAnnotations